---
title: "IBD Network Based Functional Prediction"
author: "Montana Kay Lara"
date: "June 16, 2019"
output: 
  html_document: default
  pdf_document: default
---

This workflow accompanies the paper "Identification of novel loci controlling IBD susceptibility by utilizing the genetic diversity of wild-derived mice." The purpose of this workflow was to prioritize genes in chromosomes 1 and 2, and the Ccc1 locus associated with IBD in mice. 

This workflow incorporates IBD mouse model immune cell transcriptomic data from the Krementsov lab and human IBD GWAS genes (de Lange KM, et al. doi:10.1038/ng.3760). Here we take the GWAS gens to train SVMs using the intestine and hemolymphoid networks from Functional Networks of Tissues in Mice (http://fntm.princeton.edu). We combine the SVM scores and expression data to rank positional candidates contributing to IBD susceptibility. 

UPDATE: June 30, 2020

## Outline

I.    Project Setup
II.   GWAS Genes and ID Conversion
III.  Expression Data
IV.   Gene Positions
V.    Data Cleanup and Consolidation
VI.   SVM
VII.  Integrate Gene Information with SVM Output
VIII. Gene Scores
IX.   Plot value preparation
X.    Plotting Integrated Ranking
XI.   Heatmap tables
XII.  Heatmaps
XIII. Ccc1 locus
XIV.  Supplementals/other

## I. Project Setup

A. Packages
```{r setup}
all.packages <- c("tidyverse", 
                  "ggplot2", 
                  "here", 
                  "readxl",
                  "conflicted", 
                  "gprofiler2", 
                  "R.utils", 
                  "doParallel", 
                  "gprofiler2", 
                  "e1071",
                  "DescTools", 
                  "knitr",
                  "igraph",
                  "magrittr", 
                  "RColorBrewer", 
                  "biomaRt",
                  "hablar", 
                  "plotly",
                  "forcats", 
                  "magrittr",
                  "rPref",
                  "lazyeval",
                  "rlang",
                  "reshape2",
                  "cowplot")
for(i in 1:length(all.packages)){library(all.packages[i], character.only = TRUE)} 
conflict_prefer("filter", "dplyr")
```

B. Directory setup
```{r setup_dir}
project.dir <- here()
code.dir <- here("Code")
data.dir <- here("Data")
results.int.dir <- here("Results_int")
results.hemo.dir <- here("Results_hemo")
figures.dir <- here("Figures")
```

C. Source the code used in the project
```{r source_code}
all.fun.dir <- list.files(code.dir, full.names = TRUE)
for(i in 1:length(all.fun.dir)){
   all.fun <- list.files(all.fun.dir[i], full.names = TRUE)
    for(j in 1:length(all.fun)){source(all.fun[j])}
    }
```

D. Download networks
Download mouse networks from functional networks of tissues in mice (FNTM). The specified networks used in this analysis were downloaded on July 25, 2019 (then again on June 30, 2020 when updating this workflow). The function download.tissue.network() will download the specified network (top edges) and place it in the specified directory
```{r download_network}
# 1. Intestine 
int.net <- here("Data", "intestine_top.Rdata")
if(!file.exists(int.net)){
    download.tissue.net(tissue = "intestine", organism = "mouse", project.dir = data.dir)
    }

# 2. Hemolymphoid
hemo.net <- here("Data", "hemolymphoid_system_top.Rdata")
if(!file.exists(hemo.net)){
    download.tissue.net(tissue = "hemolymphoid system", organism = "mouse", project.dir = data.dir)
    }
```

E. Read in the downlaoded function genomic network
```{r read_fgn}
# 1. Intestine
intfgn <- readRDS(here("Data", "intestine_top.Rdata"))

# 2. Hemolymphoid
hemofgn <- readRDS(here("Data", "hemolymphoid_system_top.Rdata"))
```

## II.   GWAS Genes and ID Conversion

```{r GWAS_genes}
# import list from excel file
IBDGWASgenes <- read_excel(here("Data", "DE_genes", "IBDGWASgenes.xlsx"), sheet = "IBD GWAS column list")

# convert GWAS genes from human to mouse using gprofiler
IBD.mouse.orth <- gorth(IBDGWASgenes[[1]], source_organism = "hsapiens", target_organism = "mmusculus")

# convert mouse genes to entrez IDs
IBD.gconvert <- gconvert(IBD.mouse.orth$ortholog_name, organism = "mmusculus", target = "ENTREZGENE_ACC")
```

## III.  Expression Data

This workflow incorporates gene expression data from five different immune cells (Krementsov lab). 
A. Read in DEG data
```{r DEG_data}
#read in excel file with differentially expressed genes (DEG) data
DEG.all.original <- read_excel(here("Data", "DE_genes", "Full microarray for Matt - v3.xlsx"), sheet = "v3 2nd dup removal")

#cleanup (NOTE: this is specific to this excel file with the DEGs, and this cleanup will have duplicates because of the nature of the expression data - ie, we won't remove duplicates at this point because lowest expression p values may be in different observations for each immune cell and cleaner filtering can happen later)
DEG.all <- DEG.all.original %>% 
              rename(APCpv    = `stepup(p-value(PWD * APCs vs. B6 * APCs))`, 
                     BCELLpv  = `stepup(p-value(PWD * Bcells vs. B6 * Bcells))`, 
                     CD4pv    = `stepup(p-value(PWD * CD4 vs. B6 * CD4))`, 
                     CD8pv    = `stepup(p-value(PWD * CD8 vs. B6 * CD8))`, 
                     TREGpv   = `stepup(p-value(PWD * Tregs vs. B6 * Tregs))`,
                     EntrezID = `Entrez Gene ID`,
                     Symbol   = `NEW Symbol`,
                     APCfc    = `Fold-Change(PWD * APCs vs. B6 * APCs)`,
                     BCELLfc  = `Fold-Change(PWD * Bcells vs. B6 * Bcells)`,
                     CD4fc    = `Fold-Change(PWD * CD4 vs. B6 * CD4)`,
                     CD8fc    = `Fold-Change(PWD * CD8 vs. B6 * CD8)`,
                     TREGfc   = `Fold-Change(PWD * Tregs vs. B6 * Tregs)`) %>% 
              select(Symbol, EntrezID, Chr, APCpv, BCELLpv, CD4pv, CD8pv, TREGpv, 
                     APCfc, BCELLfc, CD4fc, CD8fc, TREGfc) %>% 
              mutate(across(starts_with("EntrezID"), as.character))
```

## IV.   Gene Positions

We used biomart to obtain the genomic position and entrez IDs for genes on the Ccc1 locus. First, we set up the mmusculus biomart libary. The library is used to retrive information about gene IDs and genomic position. The function getBM requires attributes to be outputed, filters for the inputed genes, values of the genes we input, and the biomart library. 
```{r Ccc1_genes}
#create mart by selecting biomart database and dataset
mmusculus <- useEnsembl(biomart="ensembl", dataset="mmusculus_gene_ensembl") 
#listAttributes(mmusculus) will give list of all potential attributes to output
#listFilters(mmusculus) will give list of all potential filters for the values

#set genomic region for value inputs
chr <- 12
region.min <- 94800000
region.max <- 112300000
chr.region <- paste(chr, region.min, region.max, sep = ":")

#get position information for all genes in the region
Ccc1.gene.biomart <- getBM(attributes = c("external_gene_name", "entrezgene_id", "start_position", "end_position", "chromosome_name"), 
                        filters = "chromosomal_region", 
                        values = chr.region, 
                        mart = mmusculus)

#make entregene_id characters and take out those genes without entrez ids (and rename column names to match DEG.all)
Ccc1.gene.info <- Ccc1.gene.biomart %>% 
  mutate(across(starts_with("entrezgene_id"), as.character)) %>% 
  filter(!is.na(entrezgene_id)) %>% 
  rename(Symbol = external_gene_name,
         EntrezID = entrezgene_id, 
         Chr = chromosome_name) %>% 
  mutate(across(starts_with("Chr"), as.character))
```

## V.    Data Cleanup and Consolidation

We consolidated gene infomration into a table (DEG.all) with gene expression data and position data. We also filtered for chromosome 1 and 2 (DEG.chr). In addition, this specific dataset has some 0 values for p-values, which will need to be converted to non-zero values to put through this workflow. 
A. Add gene position table with DEG table
```{r DEG.all}
#add Ccc1 to DEG (there are some in Ccc1 entrez genes not in Deg, so filtering later would miss genes)
DEG.all <- bind_rows(DEG.all, Ccc1.gene.info)
```

B. Chromosome 1 and 2 filter
```{r Chr_filter}
#filter chr 1 and 2 (this will make running score.candidates go faster, you can also do all DEG.all)
DEG.chr <- DEG.all %>% 
  filter(Chr == 1 | Chr == 2)
```

C. DEG p-value cleanup
Convert p-values that are 0 to non-zero minimums (to be able to log transform and scale)
```{r DEG_cleanup}
#Finding min non-zero value for each p-value (ONLY from DEG.chr, did not go through entire DEG.all), this may not be necessary for each p-value column
min(DEG.chr$APCpv[DEG.chr$APCpv!=0])
min(DEG.chr$BCELLpv[DEG.chr$BCELLpv!=0])
min(DEG.chr$CD4pv[DEG.chr$CD4pv!=0])
min(DEG.chr$CD8pv[DEG.chr$CD8pv!=0])
min(DEG.chr$TREGpv[DEG.chr$TREGpv!=0])

#replacing 0 values with non-zero mins
DEG.chr %<>%
  mutate(APCpv   = replace(APCpv, which(APCpv == 0), 10^(-43)),
         BCELLpv = replace(BCELLpv, which(BCELLpv == 0), 10^(-41)),
         CD4pv   = replace(CD4pv, which(CD4pv == 0), 10^(-28)), 
         CD8pv   = replace(CD8pv, which(CD8pv == 0), 10^(-28)), 
         TREGpv  = replace(TREGpv, which(TREGpv == 0), 10^(-35)))
```

## VI. SVM

We ran the SVM analysis separately for the specified networks. The SVMs were trained to distinguish genes in the gene list from randomly chosen genes from outside the list. We use the function generate.triage.models() to accomplish this, which uses the ``svm`` function in the package ``e1071``. The features used by the SVM are the edge weights between all genes and the genes in the GWAS gene list. For each of the 100 SVM models, we used the GWAS genes as the true positives, while the true negatives were selected at random from outside of that gene set. Each true negative set was selected to have the same number of genes as were in the true positive set. 

A. Train SVM using the GWAS genes and the specified functional network
```{r train_SVM}
#set paramaters
n.SVM <- 100
n.cores = detectCores()-1 #The number of cores to use for parallel processing

# 1. Intestine
generate.triage.models(path = results.int.dir, 
                       project.name = "IBDint", 
                       trait.genes = IBD.gconvert$target,
                       fgn = intfgn, 
                       n.trials = n.SVM, 
                       cluster.modules = FALSE, 
                       use.SVD = FALSE, 
                       verbose = TRUE, 
                       n.cores = n.cores)

# 2. Hemolymphoid
generate.triage.models(path = results.hemo.dir, 
                       project.name = "IBDhemo", 
                       trait.genes = IBD.gconvert$target,
                       fgn = hemofgn, 
                       n.trials = n.SVM, 
                       cluster.modules = FALSE, 
                       use.SVD = FALSE, 
                       verbose = TRUE, 
                       n.cores = n.cores)
```

B. ROC curves 
Following training of the SVM models, we evaluted them by generating ROC curves to test how well the models separate genes annotated to the network from genes in the rest of the genome. To do this, use the svm.ROC function, which plots the false positive rate against the true positive rate of the SVM model across the full range of prediction values. An area under the ROC curve (AUC) indicates how well the model does overall (AUC 0.5 means model performs at random chance, AUC 1 means model has perfect classification)
```{r ROC_curve}
#1. Intestine
svm.ROC(results.int.dir, n.cores = n.cores, plot.results = TRUE)

#2. Hemolymphoid
svm.ROC(results.hemo.dir, n.cores = n.cores, plot.results = TRUE)
```

C. Apply the trained models to genes
After training the SVM models, we use them to classify each gene, where each gene recieves a score from each SVM indicating whether the model identifies it as part of the [network] gene list (a positive score) or not part of the gene list (negative score). The score.candidates function writes out all of the values to two csv tables (Candidate.Gene.SVM.Scores.csv and Candidate.Gene.FP.Rates.csv). [Note: The function also plots boxplots of the distributions of the SVM scores and the false positive (FP) rates for each gene in two jpgs (Candidate.Gene.SVM.Scores.jpg, and Candidate.Gene.FP.Rates.jpg). Plots are ordered by mean value. however, this iteration doesn't create them??]
The genes that are on the left are classified confidently by the SVM as being related to the [network] gene list. The distributions of their SVM scores do not cross 0. The genes on the right hand side are classified confidently by the SVM as not being related to the [network] gene list. The genes in between are less confidently classified across the SVM trials. 
```{r score_candidates}
#1. Intestine
score.candidates(results.int.dir, candidate.genes = DEG.chr$EntrezID)

#2. Hemolymphoid
score.candidates(results.hemo.dir, candidate.genes = DEG.chr$EntrezID)
```

## VII.   Integrate Gene Information with SVM Output

We integrated the SVM scores and FP Rates with the original gene information (DEG.all). The function merge.gene.svm.fp takes the original gene table (DEG.all) and adds the columns for the mean SVM score and mean FP rate to the genes. It writes out a table for each module with the information for each gene, including the original information from the gene table (entrez ID, gene name, expression p-values, position data, etc.), SVM score, and FP rate (Candidate.Gene.Results.rds). 
```{r merge_info}
#gene.column.name is the column name from the gene.info.table with the entrez IDs

#1. Intestine
int.final.table <- merge.gene.svm.fp(results.int.dir, gene.info.table = DEG.chr, gene.column.name = "EntrezID")

#2. Hemolymphoid
hemo.final.table <- merge.gene.svm.fp(results.hemo.dir, gene.info.table = DEG.chr, gene.column.name = "EntrezID")
```

## VIII.  Gene Scores

Each gene is scored based on the integration of SVM false positive (FP) rates, GWAS p-values, and potentially other metrics given the dataset (e.g., expression data). Given two variables (i.e., -log p-value and -log SVM FP rate), the score is based on the number of genes in the area covered by its x,y position. An individual gene with a lot of genes covered in their area will be ranked higher than a gene with fewer. Given only positional information and SVM output, genes can be ranked on their FP rates and mapped by their position (and significant GWAS genes can also be designated from the rest). 

A. Log transformation
```{r log_transformation}
#Here, there are five p-values corresponding to five cell types for this DEG dataset. Each will be log transformed and scaled to 1. 

#1. Intestine
int.final.table %<>%
  mutate(APC.log_p    = -log10(APCpv), 
         BCELL.log_p  = -log10(BCELLpv),
         CD4.log_p    = -log10(CD4pv), 
         CD8.log_p    = -log10(CD8pv), 
         TREG.log_p   = -log10(TREGpv),
         #FP rates
         log_fp = -log10(mean.fp.rate))

#2. Hemolymphoid 
hemo.final.table %<>%
  mutate(APC.log_p    = -log10(APCpv), 
         BCELL.log_p  = -log10(BCELLpv),
         CD4.log_p    = -log10(CD4pv), 
         CD8.log_p    = -log10(CD8pv), 
         TREG.log_p   = -log10(TREGpv),
         #FP rates
         log_fp = -log10(mean.fp.rate))
```

B. Score calculation
Final DEG/SVM prioritization gene score is calculated as follows: for every gene in the -log p-value x -log SVM FP rate plot, how many genes are in the are covered by its position. Individual genes with more (other) genes in their positional box will be ranked higher as they dominate more of the genes contained in the plot. This calculation maximizes both measures for a single gene. (note: this score incorporates non-significant values in the chromosome, but the plots will be significant genes only)
```{r score}
#the "total.score" value will be the number of genes that the individual gene performs better than in both p-value and FP rate (for all the genes in the "final.table," which have been previously filtered by chromosome 1 and 2)
#1. Intestine
int.final.table %<>% 
  rowwise() %>% 
  mutate(APC.total.score   = sum(APC.log_p > .$APC.log_p & log_fp > .$log_fp),
         BCELL.total.score = sum(BCELL.log_p > .$BCELL.log_p & log_fp > .$log_fp),
         CD4.total.score   = sum(CD4.log_p > .$CD4.log_p & log_fp > .$log_fp), 
         CD8.total.score   = sum(CD8.log_p > .$CD8.log_p & log_fp > .$log_fp), 
         TREG.total.score  = sum(TREG.log_p > .$TREG.log_p & log_fp > .$log_fp)) %>% 
  ungroup()

#2. Hemolymphoid
hemo.final.table %<>% 
  rowwise() %>% 
  mutate(APC.total.score   = sum(APC.log_p > .$APC.log_p & log_fp > .$log_fp),
         BCELL.total.score = sum(BCELL.log_p > .$BCELL.log_p & log_fp > .$log_fp),
         CD4.total.score   = sum(CD4.log_p > .$CD4.log_p & log_fp > .$log_fp), 
         CD8.total.score   = sum(CD8.log_p > .$CD8.log_p & log_fp > .$log_fp), 
         TREG.total.score  = sum(TREG.log_p > .$TREG.log_p & log_fp > .$log_fp)) %>% 
  ungroup()
```

## IX.   Plot value preparation

The following tables were created with various filters and values to be plotted. Final scores have already been calculated, and any filtering done here is for visualization purposes only. Each cell type requires its own table to be able to scale properly
A. Filter significant p-values for each cell type, and scale log values for each SIG table
```{r sigFilter_and_scale}
# 1. Intestine
APC.SIG.int.final.table <- IBD.sigfilter.scale(final.table = int.final.table, pval.col = "APCpv", logpv.col = "APC.log_p")
BCELL.SIG.int.final.table <- IBD.sigfilter.scale(final.table = int.final.table, pval.col = "BCELLpv", logpv.col = "BCELL.log_p")
CD4.SIG.int.final.table <- IBD.sigfilter.scale(final.table = int.final.table, pval.col = "CD4pv", logpv.col = "CD4.log_p")
CD8.SIG.int.final.table <- IBD.sigfilter.scale(final.table = int.final.table, pval.col = "CD8pv", logpv.col = "CD8.log_p")
TREG.SIG.int.final.table <- IBD.sigfilter.scale(final.table = int.final.table, pval.col = "TREGpv", logpv.col = "TREG.log_p")

# 2. Hemolymphoid
APC.SIG.hemo.final.table <- IBD.sigfilter.scale(final.table = hemo.final.table, pval.col = "APCpv", logpv.col = "APC.log_p")
BCELL.SIG.hemo.final.table <- IBD.sigfilter.scale(final.table = hemo.final.table, pval.col = "BCELLpv", logpv.col = "BCELL.log_p")
CD4.SIG.hemo.final.table <- IBD.sigfilter.scale(final.table = hemo.final.table, pval.col = "CD4pv", logpv.col = "CD4.log_p")
CD8.SIG.hemo.final.table <- IBD.sigfilter.scale(final.table = hemo.final.table, pval.col = "CD8pv", logpv.col = "CD8.log_p")
TREG.SIG.hemo.final.table <- IBD.sigfilter.scale(final.table = hemo.final.table, pval.col = "TREGpv", logpv.col = "TREG.log_p")
```

B. Pareto front and top genes for each cell type
```{r Pareto_genes}
#1. Intestine
APC.int.pref <- psel(APC.SIG.int.final.table, high(log_p1) * (high(log_fp1)))
BCELL.int.pref <- psel(BCELL.SIG.int.final.table, high(log_p1) * (high(log_fp1)))
CD4.int.pref <- psel(CD4.SIG.int.final.table, high(log_p1) * (high(log_fp1)))
CD8.int.pref <- psel(CD8.SIG.int.final.table, high(log_p1) * (high(log_fp1)))
TREG.int.pref <- psel(TREG.SIG.int.final.table, high(log_p1) * (high(log_fp1)))

#2. Hemolymphoid 
APC.hemo.pref <- psel(APC.SIG.hemo.final.table, high(log_p1) * (high(log_fp1)))
BCELL.hemo.pref <- psel(BCELL.SIG.hemo.final.table, high(log_p1) * (high(log_fp1)))
CD4.hemo.pref <- psel(CD4.SIG.hemo.final.table, high(log_p1) * (high(log_fp1)))
CD8.hemo.pref <- psel(CD8.SIG.hemo.final.table, high(log_p1) * (high(log_fp1)))
TREG.hemo.pref <- psel(TREG.SIG.hemo.final.table, high(log_p1) * (high(log_fp1)))
```

```{r top10_genes}
#1. Intestine
APC.int.top10 <- APC.SIG.int.final.table %>% slice_max(APC.total.score, n = 10)
BCELL.int.top10 <- BCELL.SIG.int.final.table %>% slice_max(BCELL.total.score, n = 10)
CD4.int.top10 <- CD4.SIG.int.final.table %>% slice_max(CD4.total.score, n = 10)
CD8.int.top10 <- CD8.SIG.int.final.table %>% slice_max(CD8.total.score, n = 10)
TREG.int.top10 <- TREG.SIG.int.final.table %>% slice_max(TREG.total.score, n = 10)

#2. Hemolymphoid 
APC.hemo.top10 <- APC.SIG.hemo.final.table %>% slice_max(APC.total.score, n = 10)
BCELL.hemo.top10 <- BCELL.SIG.hemo.final.table %>% slice_max(BCELL.total.score, n = 10)
CD4.hemo.top10 <- CD4.SIG.hemo.final.table %>% slice_max(CD4.total.score, n = 10)
CD8.hemo.top10 <- CD8.SIG.hemo.final.table %>% slice_max(CD8.total.score, n = 10)
TREG.hemo.top10 <- TREG.SIG.hemo.final.table %>% slice_max(TREG.total.score, n = 10)
```

## X.    Plotting Integrated Ranking

1. Intestine
```{r APC_int_plot}
IBD_gene_plotr(SIG.final.table = APC.SIG.int.final.table, pref = APC.int.pref, top10 = APC.int.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "APC")
ggsave("APC_int.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

```{r BCELL_int_plot}
IBD_gene_plotr(SIG.final.table = BCELL.SIG.int.final.table, pref = BCELL.int.pref, top10 = BCELL.int.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "BCELL")
ggsave("BCELL_int.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

```{r CD4_int_plot}
IBD_gene_plotr(SIG.final.table = CD4.SIG.int.final.table, pref = CD4.int.pref, top10 = CD4.int.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "CD4")
ggsave("CD4_int.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

```{r CD8_int_plot}
IBD_gene_plotr(SIG.final.table = CD8.SIG.int.final.table, pref = CD8.int.pref, top10 = CD8.int.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "CD8")
ggsave("CD8_int.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

```{r TREG_int_plot}
IBD_gene_plotr(SIG.final.table = TREG.SIG.int.final.table, pref = TREG.int.pref, top10 = TREG.int.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "TREG")
ggsave("TREG_int.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

2. Hemolymphoid
```{r APC_hemo_plot}
IBD_gene_plotr(SIG.final.table = APC.SIG.hemo.final.table, pref = APC.hemo.pref, top10 = APC.hemo.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "APC")
ggsave("APC_hemo.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

```{r BCELL_hemo_plot}
IBD_gene_plotr(SIG.final.table = BCELL.SIG.hemo.final.table, pref = BCELL.hemo.pref, top10 = BCELL.hemo.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "BCELL")
ggsave("BCELL_hemo.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

```{r CD4_hemo_plot}
IBD_gene_plotr(SIG.final.table = CD4.SIG.hemo.final.table, pref = CD4.hemo.pref, top10 = CD4.hemo.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "CD4")
ggsave("CD4_hemo.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

```{r CD8_hemo_plot}
IBD_gene_plotr(SIG.final.table = CD8.SIG.hemo.final.table, pref = CD8.hemo.pref, top10 = CD8.hemo.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "CD8")
ggsave("CD8_hemo.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

```{r TREG_hemo_plot}
IBD_gene_plotr(SIG.final.table = TREG.SIG.hemo.final.table, pref = TREG.hemo.pref, top10 = TREG.hemo.top10, 
               data.genename.col = "Symbol", pref.genename.col = "Symbol", top10.genename.col = "Symbol", 
               x.pv.col = "log_p1", y.fpr.col = "log_fp1", title = "TREG")
ggsave("TREG_hemo.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

## XI. Heatmap tables

We created heatmaps of differential gene expression for chromosome 1 and 2 for each network SVM, as well as the overlap of significant genes and GWAS genes for both chromosomes (1:1).
A. Create tables with all values for heatmap
```{r heatmap_tables}
#1. Intestine
chr1.int.hm.table <- int.final.table %>% 
  filter(Chr == 1) %>% 
  filter(Symbol %in% c(APC.int.pref$Symbol, BCELL.int.pref$Symbol, CD4.int.pref$Symbol, CD8.int.pref$Symbol, TREG.int.pref$Symbol, APC.int.top10$Symbol, BCELL.int.top10$Symbol, CD4.int.top10$Symbol, CD8.int.top10$Symbol, TREG.int.top10$Symbol)) 


chr2.int.hm.table <- int.final.table %>% 
  filter(Chr == 2) %>% 
  filter(Symbol %in% c(APC.int.pref$Symbol, BCELL.int.pref$Symbol, CD4.int.pref$Symbol, CD8.int.pref$Symbol, TREG.int.pref$Symbol, APC.int.top10$Symbol, BCELL.int.top10$Symbol, CD4.int.top10$Symbol, CD8.int.top10$Symbol, TREG.int.top10$Symbol))

#2. Hemolymphoid
chr1.hemo.hm.table <- hemo.final.table %>% 
  filter(Chr == 1) %>% 
  filter(Symbol %in% c(APC.hemo.pref$Symbol, BCELL.hemo.pref$Symbol, CD4.hemo.pref$Symbol, CD8.hemo.pref$Symbol, TREG.hemo.pref$Symbol, APC.hemo.top10$Symbol, BCELL.hemo.top10$Symbol, CD4.hemo.top10$Symbol, CD8.hemo.top10$Symbol, TREG.hemo.top10$Symbol)) 


chr2.hemo.hm.table <- hemo.final.table %>% 
  filter(Chr == 2) %>% 
  filter(Symbol %in% c(APC.hemo.pref$Symbol, BCELL.hemo.pref$Symbol, CD4.hemo.pref$Symbol, CD8.hemo.pref$Symbol, TREG.hemo.pref$Symbol, APC.hemo.top10$Symbol, BCELL.hemo.top10$Symbol, CD4.hemo.top10$Symbol, CD8.hemo.top10$Symbol, TREG.hemo.top10$Symbol))
```

B. Order by highest total.score and table melt
```{r order_and_melt}
chr1.int.melt <- IBD_order_melt(chr1.int.hm.table)
chr2.int.melt <- IBD_order_melt(chr2.int.hm.table)
chr1.hemo.melt <- IBD_order_melt(chr1.hemo.hm.table)
chr2.hemo.melt <- IBD_order_melt(chr2.hemo.hm.table)
```

C. 1:1 heatmap tables
```{r one_to_one_table}
#IBDGWAS 1:1 genes (from DK)
chr.1to1 <- tibble(Symbol = c("Atp1b1", "Cacna1s", "Cd28", "F5", "Gpr35", "Nme7", "Ptprc", "Slamf8", "Ada", "Asxl1", "Cd40", "Il2ra", "Itga4", "Kif3b", "Pkig", "Serinc3", "Tm9sf4", "Ttpal"),
                   chr = c(1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2))
chr.1to1 <- right_join(int.final.table, chr.1to1, by = "Symbol")
#restructure and melt
chr1.1to1 <- chr.1to1 %>% filter(chr == 1)
chr2.1to1 <- chr.1to1 %>% filter(chr == 2)
chr1.1to1.melt <- IBD_order_melt(chr1.1to1)
chr2.1to1.melt <- IBD_1to1_melt(chr2.1to1)
```

## XII. Heatmaps

A. SVM heatmaps
```{r SVM_heatmap}
int_chr1_HM <- INT.chr1.heatmap(chr1.int.melt)
int_chr2_HM <- INT.chr2.heatmap(chr2.int.melt)
hemo_chr1_HM <- HEMO.chr1.heatmap(chr1.hemo.melt)
hemo_chr2_HM <- HEMO.chr2.heatmap(chr2.hemo.melt)
```

B. 1:1 heatmaps
```{r one_to_one_heatmap}
chr1_1to1_HM <- chr1.1to1.heatmap(chr1.1to1.melt)
chr2_1to1_HM <- chr2.1to1.heatmap(chr2.1to1.melt)
```

C. Sizing and printing heatmaps
```{r heatmap_save}
#make heatmaps have same plotsize width
ap <- align_plots(int_chr1_HM, int_chr2_HM, hemo_chr1_HM, hemo_chr2_HM, chr1_1to1_HM, chr2_1to1_HM, align = "v")

ggdraw(ap[[1]])
ggsave("int_chr1_heatmap.svg", path = figures.dir, height = 10, width = 10, units = "in")
ggdraw(ap[[2]])
ggsave("int_chr2_heatmap.svg", path = figures.dir, height = 10, width = 10, units = "in")
ggdraw(ap[[3]])
ggsave("hemo_chr1_heatmap.svg", path = figures.dir, height = 10, width = 10, units = "in")
ggdraw(ap[[4]])
ggsave("hemo_chr2_heatmap.svg", path = figures.dir, height = 10, width = 10, units = "in")
ggdraw(ap[[5]])
ggsave("chr1_1to1_heatmap.svg", path = figures.dir, height = 10, width = 10, units = "in")
ggdraw(ap[[6]])
ggsave("chr2_1to1_heatmap.svg", path = figures.dir, height = 10, width = 10, units = "in")
```

## Supplemental/other

```{r other_sup}
top.total <- bind_rows(chr1.int.hm.table, chr2.int.hm.table, chr1.hemo.hm.table, chr2.hemo.hm.table)
top.in.training <- inner_join(top.total, IBD.gconvert, by = c("Symbol" = "input"))
final.table.combined <- bind_rows(int.final.table, hemo.final.table)
```

## Ccc1 locus

In order to evaluate the Ccc1 locus, we had to copy the int and hemo results into new folders to be able to use "score.candidates" again using the Ccc1 locus genes
A. Score candidates 
```{r score_candidates_ccc1}
results.Ccc1.int.dir <- here("Results_Ccc1_int")
results.Ccc1.hemo.dir <- here("Results_Ccc1_hemo")

DEG.Ccc1 <- inner_join(DEG.all, Ccc1.gene.info, by = "Symbol")

#1. Intestine
score.candidates(results.Ccc1.int.dir, candidate.genes = DEG.Ccc1$EntrezID.x)
#2. Hemolymphoid
score.candidates(results.Ccc1.hemo.dir, candidate.genes = DEG.Ccc1$EntrezID.x)
```

B. Merge tables
```{r merge_Ccc1}
#1. Intestine
int.Ccc1.final.table <- merge.gene.svm.fp(results.Ccc1.int.dir, gene.info.table = DEG.Ccc1, gene.column.name = "EntrezID.x")
#2. Hemolymphoid
hemo.Ccc1.final.table <- merge.gene.svm.fp(results.Ccc1.hemo.dir, gene.info.table = DEG.Ccc1, gene.column.name = "EntrezID.x")
```

```{r}
DEG.all
Ccc1.gene.info
```


